test/ziggurat/messaging/rabbitmq/producer_test.clj

```diff
@@ -99,7 +101,21 @@
                                      (reset! nippy-called? true))
                                    serialized-message)]
         (is (thrown? Exception (rm-prod/publish nil exchange-name message-payload expiration))))
-      (is (true? @nippy-called?)))))
+      (is (true? @nippy-called?))))
+
+  (testing "when lb/publish function raises an exception, it goes infinitely"
+    (let [serialized-message              (byte-array 1234)
+          exchange-name                   "exchange-test"
+          headers                         (map #(reify
+                                                  Header
+                                                  (key [_] (str "foo-" %))
+                                                  (value [_] serialized-message)) (range 1 3))
+          message-payload                 {:foo "bar" :headers headers}
+          expiration                      nil]
+      (with-redefs [lch/open                (fn [^Connection _] (create-mock-channel))
+                    lb/publish              (fn [^Channel _ ^String _ ^String _ _ _]
+                                              (throw (AlreadyClosedException. (ShutdownSignalException. true true nil nil))))]
+        (is (thrown? Exception (rm-prod/publish nil exchange-name message-payload expiration)))))))
 
```

Comment:
One more test should be added to verify the behaviour when a non-network exception is thrown.

Reply:
That has been tested in the previous test